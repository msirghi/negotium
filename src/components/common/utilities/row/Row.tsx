import { FC } from 'react';
import { ROW_DIRECTION } from '../../../../common/constants/enums';

type Props = {
  alignVerticalCenter?: boolean;
  alignHorizontalCenter?: boolean;
  className?: string;
  fullWidth?: boolean;
  onClick?: () => void;
  direction?: ROW_DIRECTION;
};

export const Row: FC<Props> = ({
  children,
  alignHorizontalCenter,
  alignVerticalCenter,
  className,
  fullWidth,
  onClick,
  direction,
}) => {
  const defaultStyles = {
    display: 'flex',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    width: 'initial',
    flexDirection: 'row',
  };

  if (alignHorizontalCenter) {
    defaultStyles.justifyContent = 'center';
  }

  if (alignVerticalCenter) {
    defaultStyles.alignItems = 'center';
  }

  if (fullWidth) {
    defaultStyles.width = '100%';
  }

  if (direction) {
    defaultStyles.flexDirection = direction;
  }

  return (
    <div
      // @ts-ignore
      style={defaultStyles}
      className={className}
      onClick={() => onClick && onClick()}
    >
      {children}
    </div>
  );
};
