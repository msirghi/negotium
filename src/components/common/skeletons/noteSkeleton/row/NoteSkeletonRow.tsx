import { useNoteSkeletonStyles } from '../styles';
import { Row } from '../../../utilities/row/Row';
import { Grid, Skeleton } from '@mui/material';

const gridItemSizes = {
  xs: 12,
  sm: 12,
  md: 6,
  lg: 4,
} as any;

export const NoteSkeletonRow = () => {
  const classes = useNoteSkeletonStyles();
  return (
    <>
      <Grid item {...gridItemSizes}>
        <Skeleton className={classes.noteSkeleton} height={240} />
      </Grid>
      <Grid item {...gridItemSizes}>
        <Skeleton className={classes.noteSkeleton} height={240} />
      </Grid>
      <Grid item {...gridItemSizes}>
        <Skeleton className={classes.noteSkeleton} height={240} />
      </Grid>
    </>
  );
};
