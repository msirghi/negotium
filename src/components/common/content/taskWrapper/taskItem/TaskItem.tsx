import { FC } from 'react';
import { ITask } from '../../../../../common/types/tasks.types';
import { Checkbox, Chip, FormControlLabel } from '@mui/material';
import styles from './TaskItem.module.scss';
import { Row } from '../../../utilities/row/Row';
import { If } from '../../../utilities/if/If';
import TaskItemUtils from './utils/utils';

type Props = {
  task: ITask;
  markAsDone: (id: ITask['id']) => void;
};

export const TaskItem: FC<Props> = ({ task, markAsDone }) => {
  const { title, dueDate } = task;

  const chipOptions = TaskItemUtils.getDateBadgeLabel(dueDate);

  return (
    <Row className={styles.tiItem} alignVerticalCenter>
      <FormControlLabel
        onChange={() => markAsDone(task.id)}
        control={<Checkbox />}
        label={title}
        sx={{ width: '%' }}
      />
      <div>
        <If condition={!!chipOptions}>
          <Chip
            label={chipOptions?.title}
            style={{
              backgroundColor: chipOptions?.backgroundColor,
              color: chipOptions?.textColor,
            }}
            size={'small'}
          />
        </If>
      </div>
    </Row>
  );
};
