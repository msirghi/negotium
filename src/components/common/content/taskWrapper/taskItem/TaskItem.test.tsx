import { TasksMock } from '../../../../../common/tests/mockData/tasks-mock';
import renderer from 'react-test-renderer';
import { TaskItem } from './TaskItem';
import { mount } from 'enzyme';
import { Checkbox } from '@mui/material';
import { act } from '@testing-library/react';
import {Row} from "../../../utilities/row/Row";

describe('TaskItem', () => {
  const defaultProps = {
    task: TasksMock[0],
    markAsDone: jest.fn(),
    onTaskSelect: jest.fn(),
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const tree = renderer.create(<TaskItem {...defaultProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('should mark task as done', () => {
    const wrapper = mount(<TaskItem {...defaultProps} />);
    const checkbox = wrapper.find(Checkbox);
    act(() => {
      // @ts-ignore
      checkbox.props().onChange({});
    });
    expect(defaultProps.markAsDone).toBeCalled();
  });

  it('should select a task', () => {
    const wrapper = mount(<TaskItem {...defaultProps} />);
    const row = wrapper.find(Row).at(0);
    act(() => {
      row.simulate('click');
    });
    expect(defaultProps.onTaskSelect).toBeCalled();
  });
});
