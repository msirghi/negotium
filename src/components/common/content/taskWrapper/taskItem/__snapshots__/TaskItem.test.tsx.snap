// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskItem should match the snapshot 1`] = `
Array [
  <div
    aria-describedby="rbd-hidden-text-0-hidden-text-0"
    className="makeStyles-container-1"
    data-rbd-drag-handle-context-id="0"
    data-rbd-drag-handle-draggable-id="uuid-1"
    data-rbd-draggable-context-id="0"
    data-rbd-draggable-id="uuid-1"
    draggable={false}
    onDragStart={[Function]}
    onTransitionEnd={null}
    role="button"
    style={
      Object {
        "transform": null,
        "transition": null,
      }
    }
    tabIndex={0}
  >
    <div>
      <div
        className="makeStyles-completedTask-3"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "width": "initial",
          }
        }
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "width": "initial",
            }
          }
        >
          <span
            className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <input
              className="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate={false}
              defaultChecked={true}
              onChange={[Function]}
              type="checkbox"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              data-testid="CheckCircleOutlineIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 7.58 10 14.17l-3.59-3.58L5 12l5 5 8-8zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
          </span>
          <div
            className="makeStyles-title-4"
          >
            [{"type":"paragraph","children":[{"text":"new task for 123 "},{"text":""}]}]
          </div>
        </div>
      </div>
      <div
        className="makeStyles-date-2"
        style={
          Object {
            "color": "#b0b0b0",
          }
        }
      >
        12 Nov, 2021
      </div>
    </div>
  </div>,
  " ",
]
`;
